### PR TITLE
Patches

### PR DESCRIPTION
**Pull request by kashav**
Patched some stuff
- JSON requests in Buildings
- Max attempts for UTSG Exams
- Use requests helper in Layers superclass

### CODE REVIEW

File: uoftscrapers/scrapers/utils/scraper.py [hunk 1]
```diff
@@ -41,6 +41,7 @@ def get(url, params=None, cookies=None, headers=None, json=False, max_attempts=1
                     doc = r
                 else:
                     sleep(0.5)
+                    attempts += 1
```
Comment (kashav) [inline uoftscrapers/scrapers/utils/scraper.py:44]: Not sure if this was by design, but whenever the request 404'd, it'd go on forever.

Try it with `UTSGExams` (any period past the current one).

Comment (qasim) [inline uoftscrapers/scrapers/utils/scraper.py:44]: good catch

Comment (arkon) [inline uoftscrapers/scrapers/utils/scraper.py:44]: Whoops, yeah. we should better handle those errors anyway. I doubt there's any reason to try again if it got a 404.

Comment (qasim) [inline uoftscrapers/scrapers/utils/scraper.py:44]: CourseFinder often 404's, 401's, or 500's. It's a real pain.

**Pull request merged**